src/pgduckdb_duckdb.cpp

```diff
@@ -136,8 +137,19 @@ DuckdbCreateConnection(List *rtables, PlannerInfo *planner_info, List *needed_co
 		}
 		pfree(duckdb_extension->data);
 	}
+}
 
-	return connection;
+duckdb::unique_ptr<duckdb::Connection>
+DuckdbCreateConnection(List *rtables, PlannerInfo *planner_info, List *needed_columns, const char *query) {
+	auto &db = DuckDBManager::Get().GetDatabase();
+	/* Add DuckDB replacement scan to read PG data */
+	auto &scans = db.instance->config.replacement_scans;
+	scans.emplace_back(pgduckdb::PostgresReplacementScan);
```

Comment:
This will add multiple `PostgresReplacementScan` into `replacement_scans` vector. Maybe add this as part of singleton construction?